### ISSUE TITLE
Occasional STOP_SENDING from client

### ISSUE THREAD

**Issue by martinthomson** [mozilla/neqo#655]
I am sometimes getting 0-RTT working, but other times the client doesn't recognize the QPACK encoder stream.  I'm getting this:

```
0s213ms TRACE [STREAM] stream 7 0->3 = [3]: 3f6102
...
0s215ms TRACE [Http3 connection] Readable stream 7.
0s215ms DEBUG Unknown stream.
```

That's the first three bytes as unrecognizable QPACK goop rather than the stream type (2 for encoder, 3 for decoder).  We have a decoder stream already (stream 11), so I'm guessing that this is the encoder stream.

This always seems to occur, even when the client is successful.  I guess it is down to timing as to whether the client is able to send the STOP_SENDING and get the RESET_STREAM back in time to affect the success of the connection.

**Comment by martinthomson** [comment]
This appears to be the result of a bad ordering between the stream type and a capacity change.  That is an HPACK integer of 128 with a 3-bit prefix of 0b001 before the stream type.  This is the result of buffering instructions in send_buf in the encoder.

I think that we should not be buffering anything on this stream.  All writes should be atomic and all attempts to insert should fail until the stream header (type + capacity instruction) is written.  Capacity changes thereafter can fail if there isn't enough space to write the instruction.

**Comment by martinthomson** [comment]
Oh, now I see how this might be my fault.  I'm doing 0-RTT so the client might be getting its settings in before the server can make the streams.  That results in buffering the capacity change instruction before the stream type.  That doesn't change the outcome: we shouldn't be buffering anything.  If we never get enough flow control to send type + capacity, then we don't need to use the dynamic table.

**Comment by ddragana** [comment]
I use buffering just because I do not want to keep states. Basically the buffer is the state. Before using the dynamic table we check if send_buf is empty and if it is not we do not use dynamic table. You can change send_buf to be a state "DynamicTableUninitialize" and "DynamicTableInitialize" but there will not be a lot of change to the code.
Sending capacity change instruction after initialization will get more complicated (WantToChangeCapacity state or something). I still prefer just keeping send_buf

**Comment by ddragana** [comment]
Fixed by #691
